arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,275 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "transform_detected_object_set.h"
+
+#include <iostream>
+#include <vital/io/camera_io.h>
+#include <vital/config/config_difference.h>
+#include <vital/util/string.h>
+#include <vital/types/bounding_box.h>
+#include <Eigen/Core>
+
+using namespace kwiver::vital;
+
+namespace kwiver {
+namespace arrows {
+namespace core {
+
+// ------------------------------------------------------------------
+transform_detected_object_set::
+transform_detected_object_set()
+  : src_camera_krtd_file_name( "" )
+  , dest_camera_krtd_file_name( "" )
+{
+}
+
+// ------------------------------------------------------------------
+transform_detected_object_set::
+transform_detected_object_set(kwiver::vital::camera_perspective_sptr src_cam,
+			      kwiver::vital::camera_perspective_sptr dest_cam)
+  : src_camera( src_cam )
+  , dest_camera( dest_cam )
+{
+}
+
+// ------------------------------------------------------------------
+vital::config_block_sptr
+transform_detected_object_set::get_configuration() const
+{
+  // Get base config from base class
+  vital::config_block_sptr config = vital::algorithm::get_configuration();
+
+  config->set_value( "src_camera_krtd_file_name", src_camera_krtd_file_name,
+                     "Source camera KRTD file name path" );
+
+  config->set_value( "dest_camera_krtd_file_name", dest_camera_krtd_file_name,
+                     "Destination camera KRTD file name path" );
+
+  return config;
+}
+
+
+// ------------------------------------------------------------------
+void
+transform_detected_object_set::
+set_configuration( vital::config_block_sptr config_in )
+{
+  vital::config_block_sptr config = this->get_configuration();
+
+  config->merge_config( config_in );
+  this->src_camera_krtd_file_name =\
+    config->get_value< std::string > ( "src_camera_krtd_file_name" );
+  this->dest_camera_krtd_file_name =\
+    config->get_value< std::string > ( "dest_camera_krtd_file_name" );
+
+  this->src_camera =\
+    kwiver::vital::read_krtd_file( this->src_camera_krtd_file_name );
+  this->dest_camera =\
+    kwiver::vital::read_krtd_file( this->dest_camera_krtd_file_name );
+}
+
+
+// ------------------------------------------------------------------
+bool
+transform_detected_object_set::
+check_configuration( vital::config_block_sptr config ) const
+{
+  kwiver::vital::config_difference cd( this->get_configuration(), config );
+  const auto key_list = cd.extra_keys();
+
+  if ( ! key_list.empty() )
+  {
+    LOG_WARN( logger(), "Additional parameters found in config block that are "
+	      "not required or desired: "
+              << kwiver::vital::join( key_list, ", " ) );
+  }
+
+  return true;
+}
+
+
+// ------------------------------------------------------------------
+vector_3d
+transform_detected_object_set::
+backproject_to_ground( const kwiver::vital::camera_perspective_sptr camera,
+		       const vector_2d img_pt ) const
```

Comment:
The indentation here is does not line up with the previous line.  You also might have some tabs here instead of spaces.  I would convert your tabs to spaces and then line up lines like this throughout the file using spaces.  It may help to tell your editor to use spaces instead of tabs for C++ files.  Most editors support that.